src/analyser.cpp

```diff
@@ -126,38 +131,54 @@ void AnalyserInternalVariable::makeVoi()
 
 void AnalyserInternalVariable::makeState()
 {
-    if (mType == Type::UNKNOWN) {
+    switch (mType) {
+    case Type::UNKNOWN:
         mType = Type::SHOULD_BE_STATE;
-    } else if (mType == Type::CONSTANT) {
+
+        break;
+    case Type::INITIALISED:
         mType = Type::STATE;
+
+        break;
+    default: // Other types we don't care about.
+        break;
     }
 }
 
+void AnalyserInternalVariable::makeConstant(size_t &index)
+{
+    mIndex = ++index;
+    mType = Type::CONSTANT;
+}
+
 struct AnalyserInternalEquation
 {
     enum struct Type
     {
-        UNKNOWN,
-        TRUE_CONSTANT,
-        VARIABLE_BASED_CONSTANT,
-        RATE,
-        ALGEBRAIC
+        UNKNOWN, // 0
+        TRUE_CONSTANT, // 1
+        VARIABLE_BASED_CONSTANT, // 2
+        ODE, // 3
+        NLA, // 4
+        ALGEBRAIC // 5
```

Comment:
If you are happy to get rid of these debugging comments these are the only other ones I have found.

Reply:
Oops, my bad! Just checked and indeed it was the only other place where I had such comments.

958413b4